src/auth/MockAuthService.js

```diff
@@ -0,0 +1,273 @@
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+import PropTypes from 'prop-types';
+import { ensureDefinedConfig } from '../utils';
+
+const userPropTypes = PropTypes.shape({
+  userId: PropTypes.string.isRequired,
+  username: PropTypes.string.isRequired,
+  roles: PropTypes.arrayOf(PropTypes.string),
+  administrator: PropTypes.boolean,
+});
+
+const optionsPropTypes = {
```

Comment:
Can we somehow leverge the optionsShape proptypes defined in https://github.com/edx/frontend-platform/blob/cf35b988872727f8096a95a38d554b9ec739c56d/src/auth/interface.js#L57 here? Are these the same?

Reply:
They're a little different - the mock one has some additional fields you can pass in to mock some user data.  I think we could try to merge them, but I'm not 💯 it's worth tying the two files together... 🤔

Reply:
Hmm okay I was just wondering about what happens in we change the interface options shape in interface.js, presuming the mock always needs to conform the the interface in interface.js. If the mock is an extension of that interface I guess a kind of merge would make sense. But if that is too much work right now, I think we can farm it off as a separate enhancement maybe...